server.js

```diff
@@ -985,6 +984,48 @@ routes['join_room'] = function (user, data) {
     room.sync(user);
 }
 
+routes['search_youtube'] = function (user, data) {
+	if(user.room === null) return;
+	var room = user.room;
+
+	var query = encodeURIComponent(data.query);
+
+
+	
+	if (typeof(data.query !== query) || query.length < 1 || query.length > 100) {
+		user.send("search_message", {data: "Search must be between 1 and 100 characters"});
+	}
+	
+        var host = "www.googleapis.com";
+        var path = "/youtube/v3/search?maxResults=5&part=snippet&q=" +
+            	query +
+            	"&key=" +
+            	YOUTUBE_API_KEY;
+
+	var results = [];
+
+	try {
```

Comment:
Can you please verify that this try/catch will handle failures from inside the callback? I'm thinking it won't but I may be wrong. Otherwise, search failures will crash the application.